src/docs/guides/axum.md

```diff
@@ -0,0 +1,191 @@
+---
+title: Deploy a Rust Axum App
+---
+
+[Axum](https://docs.rs/axum/latest/axum) is a web framework for Rust that focuses on ergonomics and modularity. It is designed to work with [tokio](https://docs.rs/tokio/1.40.0/x86_64-unknown-linux-gnu/tokio/index.html) and [hyper](https://docs.rs/hyper/1.4.1/x86_64-unknown-linux-gnu/hyper/index.html).
+
+This guide covers how to deploy an Axum app to Railway in four ways:
+
+1. [One-click deploy from a template](#one-click-deploy-from-a-template).
+2. [From a GitHub repository](#deploy-from-a-github-repo).
+3. [Using the CLI](#deploy-from-the-cli).
+4. [Using a Dockerfile](#use-a-dockerfile).
+
+Now, let's create an Axum app! 🚀
+
+## Create an Axum App
+
+**Note:** If you already have an Axum app locally or on GitHub, you can skip this step and go straight to the [Deploy Axum App to Railway](#deploy-the-axum-app-to-railway).
+
+To create a new Axum app, ensure that you have [Rust](https://www.rust-lang.org/tools/install) installed on your machine.
+
+Run the following command in your terminal to create a new Axum app:
+
+```bash
+cargo new helloworld --bin
+```
+
+The command creates a new binary-based Cargo project in a `helloworld` directory.
+
+Next, `cd` into the directory and add `axum` and `tokio` as a dependency in your `Cargo.toml`:
+
+```bash
+[dependencies]
+axum = "0.7.7"
+tokio = { version = "1.41.0", features = ["full"] }
+```
+
+These dependencies are required to create a bare minimum axum application.
+
+### Modify the Application File
+
+Next, open the app in your IDE and navigate to the `src/main.rs` file.
+
+Replace the content with the code below:
+
+```rust
+use axum::{
+    routing::get,
+    Router,
+};
+
+#[tokio::main]
+async fn main() {
+    // build our application with a single route
+    let app = Router::new().route("/", get(root));
+
+    // run our app with hyper, listening globally on port 3000
+    let listener = tokio::net::TcpListener::bind("0.0.0.0:3000").await.unwrap();
```

Comment:
The application should listen on `$PORT`

Reply:
It's already listening on a default PORT of 3000. So is there something I am missing? 🤔

Reply:
This part of the guide is for the reader to run it locally which means they can use any port. 

For deployment, it ignores it and uses the default port on the V2 runtime. So does it still make sense to emphasize changing it here when it really doesn't matter?

Reply:
I don't know rust, but how does it listen on $PORT when on Railway?

But yes, you would want to do something like what you do in node -

`process.env.PORT || 3000`

Here is how the template does it -

https://github.com/railwayapp-templates/rust-starter/blob/main/src%2Fmain.rs#L36

Reply:
Oh I get you, yeah that makes sense. I will make some changes.

Reply:
Fixed.